packages/files/src/main.rs

```diff
@@ -93,8 +125,31 @@ fn transform_image(input: Value, to: &str) {
             match fs::read(path) {
                 Ok(contents) => {
                     let encoded: String = general_purpose::STANDARD_NO_PAD.encode(contents);
-                    let html = format!("<img src=\"data:image/png;base64, {encoded} \"/>");
-                    let json = json!({"name": "raw", "data": html}).to_string();
+                    let width = input["arguments"]["width"].as_f64().unwrap();
+                    let caption = input["arguments"]["caption"].as_str().unwrap();
+                    let label = input["arguments"]["label"].as_str().unwrap();
+
+                    let id = if label.is_empty() {
+                        String::new()
+                    } else {
+                        format!("id=\"{label}\"")
+                    };
+
+                    let percentage = (width * 100.0).round() as i32;
+                    let style = format!("style=\"width:{percentage}%\"");
+
+                    let mut v = vec![];
+
+                    v.push(String::from("<figure>"));
+                    v.push(format!(
+                        "<img src=\"data:image/png;base64,{encoded}\" {id} {style}/>"
```

Comment:
Maybe we should default to to just do `<img src="path_to_image"/>` instead of embedding it? But we can deal with this later too!

Reply:
Maybe I'm missing something obvious but how would the browser/HTML document find the file? I discussed creating JavaScript blobs with Jonathan a few days ago, but that seems rather messy to me.

Reply:
They don't ; )

Most markup languages just links to file path and leaves the user responsible that those files are present when viewing the transpiled document too. But, I think it' great that we support embedding the images too!

Reply:
I see. In that case it seems nice to add an embed argument, however that once again runs into the issue of having different arguments for HTML and LaTeX. Maybe that's fine?

Reply:
yeah, it's rather unfortunate but I guess the most important part is ensuring that the document compiles regardless of the output, so if you add a extra argument it should exist for latex too and just be a no-op.